supervisor/options.py

```diff
@@ -1278,7 +1278,7 @@ def set_uid(self):
             return None
         msg = self.dropPrivileges(self.uid)
         if msg is None:
-            return 'Set uid to user %s' % self.uid
+            self.logger.info('Set uid to user %s' % self.uid)
```

Comment:
`self.logger` can't be used in `set_uid()` because logging hasn't been set up yet when this function is called.  That is why `set_uid()` and `dropPrivileges()` both return the error message in a string.  Below is what happens when this patch is run.

```
# whoami 
root 

# cat supervisord.conf 
[supervisord]
user = supervisor ;valid user on this system

# supervisord -n -c supervisord.conf
Traceback (most recent call last):
  File "/usr/local/bin/supervisord", line 11, in <module>
    load_entry_point('supervisor', 'console_scripts', 'supervisord')()
  File "/path/to/supervisor/supervisor/supervisord.py", line 359, in main
    go(options)
  File "/path/to/supervisor/supervisor/supervisord.py", line 369, in go
    d.main()
  File "/path/to/supervisor/supervisor/supervisord.py", line 64, in main
    setuid_msg = self.options.set_uid()
  File "/path/to/supervisor/supervisor/options.py", line 1281, in set_uid
    self.logger.info('Set uid to user %s' % self.uid)
AttributeError: ServerOptions instance has no attribute 'logger'
```